### ISSUE TITLE
Upgrade to Argo-rollouts v2 fails

### ISSUE THREAD

**Issue by derjust** [argoproj/argo-helm#970]
### Describe the bug

```
$ helm upgrade --force --namespace=shared argo-rollouts .
Error: UPGRADE FAILED: failed to replace object: Deployment.apps "argo-rollouts" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"rollouts-controller", "app.kubernetes.io/instance":"argo-rollouts", "app.kubernetes.io/name":"argo-rollouts"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

### Related helm chart

argo-rollouts

### To Reproduce

1. Have an argo 1.0.4 running
2. Try to upgrade to argo 1.1.0 via argo-helm 2.0

### Expected behavior

The upgrade 'just works' and migrates to the new version

**Comment by mkilchhofer** [comment]
Hi @derjust 
Did you read the upgrading notes? This is an expected behavior from our side and we put some information in the README to help users upgrade:
https://github.com/argoproj/argo-helm/tree/master/charts/argo-rollouts#to-200

> ## Upgrading
> 
> ### To 2.0.0
> 
> * The argo-rollouts dashboard is added to the template and can be enabled by setting `dashboard.enabled=true`.
> * There is a breaking change where the selector label `app.kubernetes.io/component: {{ .Values.controller.component }}` is added to rollout's deployment and service in order to distinguish between the controller and the dashboard component.
>   To upgrade an existing installation, please **add the `--force` parameter** to the `helm upgrade` command or **delete the Deployment and Service resource** before you upgrade. This is necessary because Deployment's label selector is immutable.

**Comment by mkilchhofer** [comment]
Oh this looks like an interesting issue. Have to read through tomorrow or so.

When you can't tolerate a small downtime, maybe there are possibilities by deleting the Deployment and using `--cascade=false`. Have to test that.

**Comment by mubarak-j** [comment]
Considering that the operation of `upgrade --force` for helm3 is still up for debate, would it make sense to hold off releases that require the usage of `--force` flag or at least for releases that changes k8s immutable resources?

**Comment by mubarak-j** [comment]
@derjust I'm curious if you found a safe approach to upgrade the helm chart? without deleting deployments managed by agro-rollouts

@mkilchhofer any insights on how o workaround this issue without deleting deployments/manifests managed by argo-rollouts ?

**Comment by derjust** [comment]
Yes; we did as described above:

1. Delete just the deployment of the Argo controller - this keeps the deployment & replication set as-is. This also keeps the ingress configuration "as is"
   * We didn't change anything on the service deployment/rollout CRD
   * From what we have seen, this approach even works with in-flight rollouts. They just 'stall' and they continue later on. Though we tried to avoid this situation
2. Run the helm upgrade
   * This re-creates the Argo control plane
3. Argo continues where it left off
    * The rollout CRDs remained as-is

We tested this approach first and we have a pretty simple service/rollout setup w/ AWS ALB ingress. YMMV esp. with different ingress approaches (even though based on how Argo works in general I would think it *should* work, too)

Definitive a workaround and a lot of 'hands on'.
After reading https://github.com/helm/helm/issues/7082 & linked tickets I can't image a quick solution for this TBH - but I'm not a Helm expert
